internal/registry/registry.go

```diff
@@ -35,6 +38,40 @@ func (r *Registry) GetBlockchain() core.BlockChain {
 	return r.blockchain
 }
 
+// SetBeaconchain sets the beaconchain to registry.
+func (r *Registry) SetBeaconchain(bc core.BlockChain) *Registry {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	r.beaconchain = bc
+	return r
+}
+
+// GetBeaconchain gets the beaconchain from registry.
+func (r *Registry) GetBeaconchain() core.BlockChain {
+	r.mu.Lock()
+	defer r.mu.Unlock()
```

Comment:
Shared mutex is required for shared state, for example:
`
r.mu.RLock()
doSomeCodeWithSharedState() // this function can be evaluated as the same time in different goroutines.
r.mu.RULock()
`

but here we don't have anything shared, we need just to return value. 

Second, if we measure performance of that function, we will measure only calls for r.mu.Lock(), r.mu.Unlock() and one assigment. I don't think that RLock works faster than Lock, because if you check assembly code, RLock has much more code, so i assume it works slower. 

3) It doesn't matter, because anyway it works so fast and we don't have millions calls per second to feel difference.